card/mod/01_core/set/all/name_validations.rb

```diff
@@ -0,0 +1,56 @@
+event :permit_codename, :validate,
+      on: :update, changed: :codename do
```

Comment:
I know you didn't change this, but I wonder if `on: :update` is correct here.  would `on: :save` make more sense?